graphix/flow/_find_gpflow.py

```diff
@@ -0,0 +1,668 @@
+
+    References
+    ----------
+    [1] Mitosek and Backens, 2024 (arXiv:2410.23439).
+    """
+
+    def __init__(self, og: OpenGraph[_M_co]) -> None:
+        """Initialize AlgebraicOpenGraph objects.
+
+        Parameters
+        ----------
+        og : OpenGraph[_M_co]
+            The open graph in its standard representation.
+        """
+        self.og = og
+        nodes = set(og.graph.nodes)
+
+        # Nodes don't need to be sorted. We do it for debugging purposes, so we can check the matrices in intermediate steps of the algorithm.
+
+        nodes_non_input = sorted(nodes - set(og.input_nodes))
+        nodes_non_output = sorted(nodes - set(og.output_nodes))
+
+        self.non_inputs = NodeIndex()
+        self.non_inputs.extend(nodes_non_input)
+
+        self.non_outputs = NodeIndex()
+        self.non_outputs.extend(nodes_non_output)
+
+        # Needs to be a deep copy because it may be modified during runtime.
+        self.non_outputs_optim = deepcopy(self.non_outputs)
+
+    @property
+    def flow_demand_matrix(self) -> MatGF2:
+        """Return the flow-demand matrix.
+
+        Returns
+        -------
+        MatGF2
+            Flow-demand matrix
+
+        Notes
+        -----
+        See Definition 3.4 and Algorithm 1 in Mitosek and Backens, 2024 (arXiv:2410.23439).
+        """
+        return self._compute_og_matrices[0]
+
+    @property
+    def order_demand_matrix(self) -> MatGF2:
+        """Return the flow-demand matrix.
+
+        Returns
+        -------
+        MatGF2
+            Order-demand matrix
+
+        Notes
+        -----
+        See Definition 3.5 and Algorithm 1 in Mitosek and Backens, 2024 (arXiv:2410.23439).
+        """
+        return self._compute_og_matrices[1]
+
+    def _compute_reduced_adj(self) -> MatGF2:
+        r"""Return the reduced adjacency matrix (RAdj) of the open graph.
+
+        Returns
+        -------
+        adj_red : MatGF2
+            Reduced adjacency matrix.
+
+        Notes
+        -----
+        The adjacency matrix of a graph :math:`Adj_G` is an :math:`n \times n` matrix.
+
+        The RAdj matrix of an open graph OG is an :math:`(n - n_O) \times (n - n_I)` submatrix of :math:`Adj_G` constructed by removing the output rows and input columns of :math:`Adj_G`.
+
+        See Definition 3.3 in Mitosek and Backens, 2024 (arXiv:2410.23439).
+        """
+        graph = self.og.graph
+        row_tags = self.non_outputs
+        col_tags = self.non_inputs
+
+        adj_red = np.zeros((len(row_tags), len(col_tags)), dtype=np.uint8).view(MatGF2)
+
+        for n1, n2 in graph.edges:
+            for u, v in ((n1, n2), (n2, n1)):
+                if u in row_tags and v in col_tags:
+                    i, j = row_tags.index(u), col_tags.index(v)
+                    adj_red[i, j] = 1
+
+        return adj_red
+
+    @cached_property
+    def _compute_og_matrices(self) -> tuple[MatGF2, MatGF2]:
```

Comment:
I would remove the verb for naming a property.

```suggestion
    def _og_matrices(self) -> tuple[MatGF2, MatGF2]:
```